File: Third Semester - Advanced Programming Methods/Lab10/decorator/PrinterTaskRunner.cs
namespace lab10.decorator;

public class PrinterTaskRunner: AbstractTaskRunner
{
    private static readonly string dateFormat = "yyyy-MM-dd HH:mm";
    
    public PrinterTaskRunner(ITaskRunner taskRunner): base(taskRunner)
    {
        
    }

    public override void ExecuteOneTask()
    {
        base.ExecuteOneTask();
        Console.WriteLine("The task is finished at: " + DateTime.Now.ToString(dateFormat));
    }
}

File: Third Semester - Advanced Programming Methods/Lab2/decorator/PrinterTaskRunner.java
package decorator;

import model.Task;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class PrinterTaskRunner extends AbstractTaskRunner {
    private static final DateTimeFormatter timeformatterr = DateTimeFormatter.ofPattern("HH:mm");

    public PrinterTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeOneTask() {
        super.executeOneTask();
        System.out.println("Task-ul s-a executat la: " + LocalDateTime.now().format(timeformatterr));
    }

}
